src/integrations/admin/background-indexing-integration.php

```diff
@@ -111,23 +125,32 @@ public function __construct(
 		Indexable_Indexing_Complete_Action $complete_indexation_action,
 		Post_Link_Indexing_Action $post_link_indexing_action,
 		Term_Link_Indexing_Action $term_link_indexing_action,
-		Indexing_Helper $indexing_helper
+		Indexing_Helper $indexing_helper,
+		Yoast_Admin_And_Dashboard_Conditional $yoast_admin_and_dashboard_conditional,
+		Get_Request_Conditional $get_request_conditional
 	) {
-		$this->post_indexation              = $post_indexation;
-		$this->term_indexation              = $term_indexation;
-		$this->post_type_archive_indexation = $post_type_archive_indexation;
-		$this->general_indexation           = $general_indexation;
-		$this->complete_indexation_action   = $complete_indexation_action;
-		$this->post_link_indexing_action    = $post_link_indexing_action;
-		$this->term_link_indexing_action    = $term_link_indexing_action;
-		$this->indexing_helper              = $indexing_helper;
+		$this->post_indexation                       = $post_indexation;
+		$this->term_indexation                       = $term_indexation;
+		$this->post_type_archive_indexation          = $post_type_archive_indexation;
+		$this->general_indexation                    = $general_indexation;
+		$this->complete_indexation_action            = $complete_indexation_action;
+		$this->post_link_indexing_action             = $post_link_indexing_action;
+		$this->term_link_indexing_action             = $term_link_indexing_action;
+		$this->indexing_helper                       = $indexing_helper;
+		$this->yoast_admin_and_dashboard_conditional = $yoast_admin_and_dashboard_conditional;
+		$this->get_request_conditional               = $get_request_conditional;
 	}
 
 	/**
 	 * Register hooks.
 	 */
 	public function register_hooks() {
-		\add_action( 'admin_init', [ $this, 'register_shutdown_indexing' ], 10 );
+		\add_action( 'admin_init', [ $this, 'register_shutdown_indexing' ] );
+		\add_action( 'Yoast\WP\SEO\index', [ $this, 'index' ] );
```

Comment:
The `index` function always calls the complete action indicating that indexing is complete. This shouldn't be the case.

We've also adjusted what complete means so I think we should adjust the complete indexing action to also update the `last_completely_indexed_versions`. Perhaps even remove the value it's currently setting and instead unify everything else to also look at that.

Reply:
As discussed on slack:
The two meanings of "complete" still differ and both have their usecase:
- `indexables_indexing_completed` has _ever_ finished a full (re)index
- `last_completely_indexed_versions` was fully complete most recently with versions XYZ